numpy/lib/tests/test_nanfunctions.py

```diff
@@ -317,26 +317,30 @@ def test_dtype_from_dtype(self):
         codes = 'efdgFDG'
         for nf, rf in zip(self.nanfuncs, self.stdfuncs):
             for c in codes:
-                tgt = rf(mat, dtype=np.dtype(c), axis=1).dtype.type
-                res = nf(mat, dtype=np.dtype(c), axis=1).dtype.type
-                assert_(res is tgt)
-                # scalar case
-                tgt = rf(mat, dtype=np.dtype(c), axis=None).dtype.type
-                res = nf(mat, dtype=np.dtype(c), axis=None).dtype.type
-                assert_(res is tgt)
+                with suppress_warnings() as sup:
```

Comment:
Hmmm, funny. The `nanstd/nanvar` code uses:
```
np.subtract(arr, avg, out=arr, casting='unsafe')
```
which uses the original arr's dtype, but `avg` already uses the new dtype. I think the spurious complex warning is the only thing that can go really bad though, because only inexact types use this code branch. So the worst that can happen is this warning and a less precise type to store the subtraction result in. If you got a nice idea how to avoid the warning, I am all ears.